Bug/random boss spawn

anshkataria opened the pull request:
# Random Spawning of Two bosses

The spawning of bosses is randomised and now two bosses will be spawned at the end of the respective boss levels.

Fixes / Closes # (issue)
#474

## Type of change

- [x] Enhancement

# How Has This Been Tested?

I have done manual testing to check that bosses spawn in random lanes.

# Checklist:

- [x] My code follows the style guidelines of this project

- [x] I have performed a self-review of my code

- [ ] I have commented my code, particularly in hard-to-understand areas

- [ ] I have made corresponding changes to the documentation

- [ ] My changes generate no new warnings

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] New and existing unit tests pass locally with my changes

- [x] Any dependent changes have been merged and published in downstream modules

kreznik-uq reviewed (changes requested):
Tested this by playing level 5. The first boss that spawned ended the wave, so a second one didn't spawn. This probably has to do with the fact that onBossDefeated will always end the wave.

I like the fact the boss spawns in a random wave now, but I don't think there's a need to spawn 2 per level (if you think its better, I'm not gonna stop you, please just fix the other stuff in that case). 

Please fix the sonar issues, most of them seem simple to fix.

anshkataria commented:
> > > Tested this by playing level 5. The first boss that spawned ended the wave, so a second one didn't spawn. This probably has to do with the fact that onBossDefeated will always end the wave.
> > > I like the fact the boss spawns in a random wave now, but I don't think there's a need to spawn 2 per level (if you think its better, I'm not gonna stop you, please just fix the other stuff in that case).
> > > Please fix the sonar issues, most of them seem simple to fix.
> > 
> > 
> > I have fixed the sonar issues, please review it now. thanks!
> 
> I've played the branch, there aren't any new bugs as far as I can see, but
> 
> You've added code that's meant to support spawning 2 bosses. This doesn't happen right now, the level ends after you kill the first one (unless I'm killing it too quickly). Consider removing this code if it's not going to do anything, because putting unused code into main hurts maintainability. Alternatively, spawn both bosses at once if you're going to leave it in.

I have removed the second boss spawning logic now.
